doc/manual/src/glossary.md

```diff
@@ -1,11 +1,95 @@
 # Glossary
 
   - [derivation]{#gloss-derivation}\
-    A description of a build task. The result of a derivation is a
-    store object. Derivations are typically specified in Nix expressions
-    using the [`derivation` primitive](language/derivations.md). These are
-    translated into low-level *store derivations* (implicitly by
-    `nix-env` and `nix-build`, or explicitly by `nix-instantiate`).
+    A [build] task declared in the [Nix language](./language/index.md) using the built-in function [`derivation`](language/derivations.md).
+
+    [derivation]: #gloss-derivation
+
+  - [build]{#gloss-build}\
+    Produce a [store object].
+
+    See [`nix-build`](./command-ref/nix-build.md).
+
+    See [`nix build`](./command-ref/new-cli/nix3-build.md) (experimental).
+
+    Nix uses a two-stage build process:
+    A Nix expression that evaluates to a [derivation] is [instantiate]d to a [store derivation], which is then [realise]d to a [store object].
+
+    ```
+    +--------------------+
+    | Nix language       |
+    | .----------------. |
+    | | Nix expression | |
+    | '----------------' |
+    |         |          |
+    |      evaluate      |
+    |         |          |             +------------------------+
+    |         V          |             | Nix store              |
+    |   .------------.   |             |  .------------------.  |
+    |   | derivation |---|-instantiate-|->| store derivation |  |
```

Comment:
```suggestion
    |   |  derived   |---|-points to---|->|     derivation   |  |
    |   |    path    |                 |  |  and output name |  |
```

Reply:
I agree, your proposed terminology would make things much clearer.

I think though, reworking the terminology would be quite a bit of effort where we should get together and sort this out. It would require rewriting the Derivation section and quite a bit of the glossary, too.

Therefore, even if it's not super precise, is it good enough as a first approximation?